### PR TITLE
Handle NPE during remote unbind if orb didn't come up

### PR DESCRIPTION
**Pull request by olendvcook**
#14885

### CODE REVIEW

**Comment by olendvcook** [comment]
#build

File: dev/com.ibm.ws.ejbcontainer.remote/src/com/ibm/ws/ejbcontainer/remote/internal/EJBRemoteRuntimeImpl.java [hunk 1]
```diff
@@ -249,17 +250,23 @@ public synchronized void unbindAll(Object bindingDataObject) {
             }
         }
 
-        for (int i = bindingData.contextNames.length - 1; i >= 0 && !hasBindings(bindingData.contexts[i]); i--) {
+        // For contextNames, if completely empty of bindings, unbind the naming context from its parent context.
+        for (int i = bindingData.contextNames.length - 1; i >= 0 && isBindingContextEmpty(bindingData.contexts[i]); i--) {
```
Comment (tkburroughs) [inline dev/com.ibm.ws.ejbcontainer.remote/src/com/ibm/ws/ejbcontainer/remote/internal/EJBRemoteRuntimeImpl.java:254]: Seems we are starting at a leaf node, and if that context is empty, we unbind it from the parent, then back up one and check if that context is null, etc... all the way back to the root.  As soon as we find one that is not empty, we stop.

However, now we will stop as soon as we find a "null" context.... but what if the parent context exists and is empty, shouldn't we back up one and check the parent as well?  If it does exist and is empty, seems we should unbind it.

File: dev/com.ibm.ws.ejbcontainer.remote/src/com/ibm/ws/ejbcontainer/remote/internal/EJBRemoteRuntimeImpl.java [hunk 2]
```diff
@@ -249,17 +250,25 @@ public synchronized void unbindAll(Object bindingDataObject) {
             }
         }
 
-        for (int i = bindingData.contextNames.length - 1; i >= 0 && !hasBindings(bindingData.contexts[i]); i--) {
-            NamingContext parentContext = i == 0 ? rootNamingContext : bindingData.contexts[i - 1];
-            unbind(parentContext, bindingData.contextNames[i]);
+        // For contextNames, if completely empty of bindings, unbind the naming context from its parent context.
+        for (int i = bindingData.contextNames.length - 1; i >= 0; i--) {
+            if (isBindingContextEmpty(bindingData.contexts[i])) {
```
Comment (tkburroughs) [inline dev/com.ibm.ws.ejbcontainer.remote/src/com/ibm/ws/ejbcontainer/remote/internal/EJBRemoteRuntimeImpl.java:255]: This is better, but inefficient.  The code used to stop at the first non-empty list, but will now keep going.  The logic here is skipped, but `isBindingContextEmpty` will continue to be called and creates several objects and creates a list.  Could we have `isBindingContextEmpty` return `Boolean`, so we could tell the difference between True, False, and null?

**Review by tkburroughs — APPROVED** [review]

**Comment by olendvcook** [comment]
#build

**Comment by olendvcook** [comment]
#run-libby-bot

**Pull request merged**